src/content_scripts/content_origin.js

```diff
@@ -1,41 +1,29 @@
-const NEW_PAYMENT_POINTER_CHECK_RATE_MS = 1000; // 1 second
-const PAYMENT_POINTER_VALIDATION_RATE_MS = 1000 * 60 * 60; // 1 hour
-let paymentPointersBeingValidated = new Set();
+const paymentPointersBeingValidated = new Set();
 
-/**
- * Content scripts can only see a "clean version" of the DOM, i.e. a version of the DOM without
- * properties which are added by JavaScript, such as document.monetization!
- * reference: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts#Content_script_environment
- *			  https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Sharing_objects_with_page_scripts
- * So we must inject some code into the JavaScript context of the current tab in order to
- * access the document.monetization object. We inject code using a script element:
- */
-const scriptEl = document.createElement('script');
-scriptEl.text = `
-	if (document.monetization) {
-		document.monetization.addEventListener('monetizationstart', (event) => {
-			document.dispatchEvent(new CustomEvent('akita_monetizationstart', { detail: event.detail }));
-		});
-
-		document.monetization.addEventListener('monetizationprogress', (event) => {
-			document.dispatchEvent(new CustomEvent('akita_monetizationprogress', { detail: event.detail }));
-		});
-
-		document.monetization.addEventListener('monetizationstop', (event) => {
-			document.dispatchEvent(new CustomEvent('akita_monetizationstop', { detail: event.detail }));
-		});
-	}
-`;
-document.body.appendChild(scriptEl);
+const iframesWithUuids = new Set();
+const iframeUuidMap = new Map();
+const paymentPointerIframeMap = new Map();
 
+// content_origin.js only runs in the top level page so it sets window.isTopLevel to true so that
+// other content scripts can check whether they are also running in the top level page.
+// See ./content_iframe.js for more info.
+window.isTopLevel = true;
 
-setExtensionIconMonetizationState(false);
 main();
 
 /**
  * Main function to initiate the application.
  */
 async function main() {
+	document.addEventListener('akita_monetizationprogress', (event) => {
+		storeDataIntoAkitaFormat(event.detail, AKITA_DATA_TYPE.PAYMENT);
+	});
+
+	setExtensionIconMonetizationState(false);
+	trackIframesByUuid();
```

Comment:
Actually doesn't need to be called here. I had it here because I wanted to start sending uuids to iframes ASAP so that they can start sending their payment pointers to the main page. BUT `isPageMonetized()` is called immediately after this inside of `trackTimeAndSetIcon()` anyways so the call in `main()` does not actually make it faster.

Reply:
Removing this call.